### ISSUE TITLE
Record syscall events like perf

### ISSUE THREAD

**Issue by bmario** [tud-zih-energy/lo2s#172]
With perf one can use wildcards to record syscalls. We should look into how it is done and mimic that logic if it is feasible.

In particular, this should be valid:

```
# lo2s -a -E raw_syscalls:*
```

**Comment by cvonelm** [comment]
I've looked into this, as I'm stuck with a pretty gross regression in the the safe counters stuff that I need some time to think about, and the way it is done is opening the raw_syscalls tracepoint event and then calling the ioctl `PERF_EVENT_IOC_SET_FILTER` to set a ftrace filter filtering the syscalls for those we want.

According to [this](https://lwn.net/Articles/370423/) ftrace filters already support globbing on their own (so we can just directly set the perf ftrace filter to `open*`)

I would personally support a syntax such as this
```
# lo2s -a -e raw_syscalls --ftrace-filter open*
```
perf only supports globbing so it can get away with the `-e raw_syscalls:[filter]` syntax but I think it would be cool for us to support  the whole ftrace filter syntax (which conveniently means that we can pass the complexity of writing correct filters directly to the user :)) and for that `-e raw_syscalls:[filter]` might get pretty ugly relatively fast.

**Comment by bmario** [comment]
I'm not too keen about adding yet another command-line argument, which is only used for one particular event. Also, I'd like to be compatible with perf for this one. Can't we instead just split of `-E raw_syscalls:[filter]` the filter and pass that directly to ftrace?

**Comment by cvonelm** [comment]
Thats totally fine

**Comment by cvonelm** [comment]
Implemented in #206.
